src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -112,18 +115,26 @@ def _get_child_data(self, area):
                         for child in area.children}
             return self.bills_results[area.name]
 
-    def _store_area_penalties(self, results_dict, area):
+    def _calculate_device_penalties(self, area):
         if len(area.children) > 0:
             return
+
         if isinstance(area.strategy, LoadHoursStrategy):
-            penalty_energy = sum(
+            return sum(
                 area.strategy.energy_requirement_Wh.get(market.time_slot, 0) / 1000.0
                 for market in self._get_past_markets_from_area(area.parent, "past_markets"))
         elif isinstance(area.strategy, PVStrategy):
-            penalty_energy = sum(
+            return sum(
                 area.strategy.state.available_energy_kWh.get(market.time_slot, 0)
                 for market in self._get_past_markets_from_area(area.parent, "past_markets"))
         else:
+            return None
+
+    def _store_area_penalties(self, results_dict, area):
```

Comment:
This will deprecated in the frame of D3ASIM-2303.
OR, you could also remove all penalties and totals_with_penalties from the "normal" bills code.

Reply:
As discussed, will do the latter. Thanks